README.md

```diff
@@ -0,0 +1,580 @@
+confusing consumers of your custom script you can override the update command
+generated at the top of requirements files by setting the
+`CUSTOM_COMPILE_COMMAND` environment variable.
+
+```console
+$ CUSTOM_COMPILE_COMMAND="./pipcompilewrapper" pip-compile requirements.in
+#
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
+#
+#    ./pipcompilewrapper
+#
+asgiref==3.6.0
+    # via django
+django==4.1.7
+    # via -r requirements.in
+sqlparse==0.4.3
+    # via django
+```
+
+### Workflow for layered requirements
+
+If you have different environments that you need to install different but
+compatible packages for, then you can create layered requirements files and use
+one layer to constrain the other.
+
+For example, if you have a Django project where you want the newest `2.1`
+release in production and when developing you want to use the Django debug
+toolbar, then you can create two `*.in` files, one for each layer:
+
+```
+# requirements.in
+django<2.2
+```
+
+At the top of the development requirements `dev-requirements.in` you use `-c
+requirements.txt` to constrain the dev requirements to packages already
+selected for production in `requirements.txt`.
+
+```
+# dev-requirements.in
+-c requirements.txt
+django-debug-toolbar<2.2
+```
+
+First, compile `requirements.txt` as usual:
+
+```
+$ pip-compile
+#
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
+#
+#    pip-compile
+#
+django==2.1.15
+    # via -r requirements.in
+pytz==2023.3
+    # via django
+```
+
+Now compile the dev requirements and the `requirements.txt` file is used as
+a constraint:
+
+```console
+$ pip-compile dev-requirements.in
+#
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
+#
+#    pip-compile dev-requirements.in
+#
+django==2.1.15
+    # via
+    #   -c requirements.txt
+    #   django-debug-toolbar
+django-debug-toolbar==2.1
+    # via -r dev-requirements.in
+pytz==2023.3
+    # via
+    #   -c requirements.txt
+    #   django
+sqlparse==0.4.3
+    # via django-debug-toolbar
+```
+
+As you can see above, even though a `2.2` release of Django is available, the
+dev requirements only include a `2.1` version of Django because they were
+constrained. Now both compiled requirements files can be installed safely in
+the dev environment.
+
+To install requirements in production stage use:
+
+```console
+$ pip-sync
+```
+
+You can install requirements in development stage by:
+
+```console
+$ pip-sync requirements.txt dev-requirements.txt
+```
+
+### Version control integration
+
+You might use `pip-compile` as a hook for the [pre-commit](https://github.com/pre-commit/pre-commit).
+See [pre-commit docs](https://pre-commit.com/) for instructions.
+Sample `.pre-commit-config.yaml`:
+
+```yaml
+repos:
+    - repo: https://github.com/jazzband/pip-tools
+    rev: 6.13.0
+    hooks:
+        - id: pip-compile
```

Comment:
YAML indentation is broken here, let's try this:
```suggestion
- repo: https://github.com/jazzband/pip-tools
  rev: 6.13.0
  hooks:
  - id: pip-compile
```

Reply:
`pre-commit` caught the indentation for us ;-)